### ISSUE TITLE
Creating attributes with arrays of strings

### ISSUE THREAD

**Issue by cmarquardt** [Unidata/netcdf4-python#592]
Is it possible to create attributes which are arrays of strings? For example,

``` python
import netCDF4 as netcdf
nc = netcdf.Dataset("seppel.nc", "w")
nc.setncattr_string("attr", ["oink", "bleek"])
nc.close()
exit()
```

Right now (python 2.7, netCDF4 1.2.4, libnetcdf 4.3.3.1), ncdump shows that rather than an array of strings, the generated attribute is a single string with a textual representation of the list:

```
netcdf seppel {

// global attributes:
            string :attr = "[\'oink\', \'bleek\']" ;
}
```

Thanks - Christian.

**Comment by jswhit** [comment]
Right now you can't create attributes that are arrays of strings.  netcdf does not have a fixed length string array type, so it would have to be done with vlen strings.

**Comment by cmarquardt** [comment]
Yes, that's what I had in mind - creating an arrays of vlen strings. Sorry, the example was a bit misleading as it had two strings of the same length. 

Would this be a useful feature request? To be honest, I haven't even tried if arrays of vlen strings can be used as an attribute with the netcdf-C; I think it can be done with Java...

**Comment by jswhit** [comment]
resolved with pull request #597
